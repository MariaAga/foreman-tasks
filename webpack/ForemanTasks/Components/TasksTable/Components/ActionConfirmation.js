import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Button } from 'patternfly-react';

export const ActionConfirmation = ({
  showModal,
  closeModal,
  title,
  message,
  onClick,
  confirmAction,
  abortAction,
}) => (
  <Modal show={showModal} onHide={closeModal}>
    <Modal.Header>
      <Button
        className="close"
        onClick={closeModal}
        aria-hidden="true"
        aria-label="Close"
      >
        &times;
      </Button>
      <Modal.Title>
        <span className="glyphicon glyphicon-warning-sign" />
        {` ${title}`}
      </Modal.Title>
    </Modal.Header>
    <Modal.Body>{message}</Modal.Body>
    <Modal.Footer>
      <Button onClick={closeModal}>{abortAction}</Button>
      <Button bsStyle="primary" onClick={onClick}>
        {confirmAction}
      </Button>
    </Modal.Footer>
  </Modal>
);

ActionConfirmation.propTypes = {
  showModal: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  confirmAction: PropTypes.string.isRequired,
  abortAction: PropTypes.string.isRequired,
  closeModal: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default ActionConfirmation;
