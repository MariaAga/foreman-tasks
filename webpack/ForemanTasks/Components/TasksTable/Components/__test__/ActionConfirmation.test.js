import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import { ActionConfirmation } from '../ActionConfirmation';

const fixtures = {
  'renders ActionConfirmation': {
    showModal: true,
    title: 'title',
    message: 'message',
    confirmAction: 'confirmAction',
    abortAction: 'abortAction',
    closeModal: () => null,
    onClick: () => null,
  },
};

describe('ActionConfirmation', () =>
  testComponentSnapshotsWithFixtures(ActionConfirmation, fixtures));
