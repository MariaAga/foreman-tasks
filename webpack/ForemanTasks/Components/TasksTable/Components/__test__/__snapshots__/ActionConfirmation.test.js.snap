// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionConfirmation renders ActionConfirmation 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <Button
      active={false}
      aria-hidden="true"
      aria-label="Close"
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="close"
      disabled={false}
      onClick={[Function]}
    >
      Ã—
    </Button>
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      <span
        className="glyphicon glyphicon-warning-sign"
      />
       title
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    message
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      abortAction
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      confirmAction
    </Button>
  </ModalFooter>
</Modal>
`;
